docusaurus/src/theme/MDXComponents/A.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import A from '@theme-original/MDXComponents/A';
+import type AType from '@theme/MDXComponents/A';
+import type {WrapperProps} from '@docusaurus/types';
+
+type Props = WrapperProps<typeof AType>;
+
+export default function AWrapper(props: Props): JSX.Element {
+  const { href } = props;
+  const isExternal =
+    href?.startsWith('http') &&
+    !href?.startsWith('https://knapsackpro.com')
+
+  return (
+    <>
+      <A
+        {...props}
+        rel={isExternal ? 'nofollow noopener noreferrer' : undefined}
+        target={isExternal ? '_blank' : undefined}
+      />
+    </>
```

Comment:
suggestion

You don't need the wrapping element:

```suggestion
    <A
      {...props}
      rel={isExternal ? 'nofollow noopener noreferrer' : undefined}
      target={isExternal ? '_blank' : undefined}
    />
```

Reply:
It was generated this way by:

```
➜  docusaurus git:(main) npm run swizzle

> docusaurus@0.0.0 swizzle
> docusaurus swizzle


    ------------------------------------------------------------------------------------------------------------------------------------------------

                                                             Update available 3.6.3 → 3.7.0

                                   To upgrade Docusaurus packages with the latest version, run the following command:
        `npm i @docusaurus/core@latest @docusaurus/plugin-client-redirects@latest @docusaurus/preset-classic@latest @docusaurus/tsconfig@latest
                                                        @docusaurus/module-type-aliases@latest`

    ------------------------------------------------------------------------------------------------------------------------------------------------

✔ Select a theme to swizzle: › @docusaurus/theme-classic
✔ Which language do you want to use? › TypeScript
✔
Select or type the component to swizzle.
* = not safe for all swizzle actions
 › MDXComponents/A (Safe)

✔ Which swizzle action do you want to do? › Wrap (Safe)
[SUCCESS]
```

The original code generated:

```typescript
import React from 'react';
import A from '@theme-original/MDXComponents/A';
import type AType from '@theme/MDXComponents/A';
import type {WrapperProps} from '@docusaurus/types';

type Props = WrapperProps<typeof AType>;

export default function AWrapper(props: Props): JSX.Element {
  return (
    <>
      <A {...props} />
    </>
  );
}
```

I'd keep it as is just to be on the safe side.